src/query/sql/src/planner/optimizer/decorrelate/decorrelate.rs

```diff
@@ -267,11 +268,20 @@ impl SubqueryRewriter {
                     &mut right_conditions,
                     &mut left_conditions,
                 )?;
+
+                let mut join_type = JoinType::LeftSingle;
+                if subquery.contain_agg.unwrap() {
```

Comment:
Is it safe to unwrap ?

Reply:
Yes, scalar subquery must have contain_agg, it was initialized by type checker.